packages/libs/deer/src/lib.rs

```diff
@@ -1,9 +1,451 @@
+        V: Visitor<'de>;
+
+    /// Deserialize a `null` (or equivalent type) value
+    ///
+    /// This type should signal missing or empty.
+    ///
+    /// Some formats do not have this concept and should in that case always error out, formats like
+    /// `JSON`, which have multiple representations for `null` should return from this function!
+    ///
+    /// # Errors
+    ///
+    /// Current value is not of type null
+    fn deserialize_null<V>(self, visitor: V) -> Result<V, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    /// Deserialize a [`bool`] value.
+    ///
+    /// Some formats might not have this concept and should always error out if that is the case.
+    ///
+    /// > **Hint**: Do not try to coerce values like `1` or `0` to booleans, this is highly
+    /// > discouraged. The [`Deserialize`] implementation or a future version of `deer` should do
+    /// > that instead.
+    ///
+    /// # Errors
+    ///
+    /// Current value is not of type bool
+    fn deserialize_bool<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    /// Deserialize a [`Number`] value.
+    ///
+    /// This is a super-type of primitive types used by Rust, this is due to the fact that some
+    /// formats (like `JSON`) do not have the concept of multiple types that represent
+    /// integers/floats.
+    ///
+    /// This method also enables a default trait implementation for all primitives types.
+    ///
+    /// > **Hint**: Do not try to coerce values like `"1"`, `"0"`, `true`, `false` to numbers, this
+    /// > is highly discouraged. The [`Deserialize`] implementation of a future version of `deer`
+    /// > would implement that functionality.
+    ///
+    /// # Errors
+    ///
+    /// Current value is not of type number
+    fn deserialize_number<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    fn deserialize_char<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    /// Deserialize a [`String`] value.
+    ///
+    /// `deer` (in contrast to [`serde`]) does not support `&str`, this is because `deer` only
+    /// handles owned data, this comes with some overhead, but enables easier implementation.
+    ///
+    /// > **Hint**: Do not try to cerce values to string, this is highly discouraged and only
+    /// > [`Deserialize`] or a future version of `deer` should implement that functionality, if
+    /// > desired.
+    ///
+    /// # Errors
+    ///
+    /// Current value is not of type string
+    ///
+    /// [`serde`]: https://serde.rs/
+    fn deserialize_string<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    fn deserialize_str<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
+
+    fn deserialize_bytes<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>;
```

Comment:
```suggestion
        V: Visitor<'de>;
 
```

Reply:
fixed in [5f0a78a](https://github.com/hashintel/hash/pull/1170/commits/5f0a78a16c1347a59d4431a4f1d548605d99568f)